src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,81 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, {
+  useCallback, useContext, useEffect, useState,
+} from 'react';
+import { FilterParams } from '../../types/filterParams';
+import { TodosContext } from '../Store/Store';
+import { getFilteredTodos } from '../../services/getfilteredTodos';
+
+export const TodoFilter = React.memo(() => {
+  const {
+    todos,
+    setFilteredTodos,
+  } = useContext(TodosContext);
+
+  const [title, setTitle] = useState('');
+  const [filter, setFilter] = useState(FilterParams.All);
+
+  const filteredTodos = useCallback(() => {
+    return getFilteredTodos(todos, filter, title);
+  }, [todos, filter, title]);
+
+  useEffect(() => {
+    setFilteredTodos(filteredTodos());
+  }, [filter, title]);
```

Comment:
you don't need second state for todos - filteredTodos.
Create state filter and query in your context. in your todolist you will assign the result of getFilteredTodos into variable and render it. every time your filter or query will change the component will rerender and variable value will updated. no need to make some side effects here